### ISSUE TITLE
CI build no longer working

### ISSUE THREAD

**Issue by uvjustin** [PyAV-Org/pyav-ffmpeg#68]
The CI build seems flaky:
 - The MacOS arm64 build fails when building openjpeg
 - The Windows build<strike>s seem to run out of memory</strike> fails
This seems odd because the test runs ran fine as of last month, and it's odd that two separate issues would pop up on different runners since then.

**Comment by jlaine** [comment]
Do you have links to failing builds?

I *think* I've already seen openjpeg failures, if memory serves the problem was flakiness with the server from which we download the code. Maybe there is a better mirror?

**Comment by uvjustin** [comment]
See https://github.com/PyAV-Org/pyav-ffmpeg/pull/69/checks with both failures

**Comment by jlaine** [comment]
ld: warning: ignoring file /usr/local/lib/liblcms2.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64

=> it looks like the build is being polluted by homebrew. Maybe try updating the `.github/workflows/build-ffmpeg.yml` to uninstall liblcms2 as we already do for various libraries

**Comment by jlaine** [comment]
I've got a tentative patch for the macOS issue, let's seen how it turns out.

The Windows failure bothers me more, it's a PITA to debug these DLL load errors.. I think I'm going to have to spin up a Windows VM.

**Comment by jlaine** [comment]
The macOS issue is fixed by my patch, but Windows is just a mess, it's failing in places it never failed before..

**Comment by jlaine** [comment]
OK, so I took the brute force approach and disabled all external libraries .. and I no longer get the error on Windows.

I then re-enabled:

- fontconfig
- gmp
- gnutls
- lzma
- zlib

.. and I again get the DLL load error, so it looks like one of these is causing our problem. Of course, that might not be the only problem but it's a start.

**Comment by jlaine** [comment]
And the culprit seems to be gnutls. Leaving all other libraries enabled and just disabling gnutls seems to be enough for the build to succeed..

**Comment by jlaine** [comment]
OK I really cannot figure out what is wrong with GnuTLS on Windows. However, I've spotted it probably doesn't even make sense to build GnuTLS for:

- MacOS: there is a native SecureTransport backend : https://github.com/FFmpeg/FFmpeg/blob/master/libavformat/tls_securetransport.c
- Windows: there is a native SChannel backend: https://github.com/FFmpeg/FFmpeg/blob/master/libavformat/tls_schannel.c

So basically it only makes sense for Linux.. and maybe not even there if we're happy to use another TLS library such as OpenSSL or even Mbed TLS. In the short term I think the easiest way forward is to build GnuTLS only and Linux and re-examine later if we want to stick to GnuTLS at all.

**Comment by uvjustin** [comment]
Great, that makes sense.
